Improve targetd plugin SSL certificate handling

tasleson opened the pull request:
Two new URI parameters have been added:

`no_ssl_verify=yes` = Disables all SSL certificate validation
`cert_file=<server certificate file >` = Certificate to use to verify server certificate

Targetd man page updated as well.

cathay4t commented on plugin/targetd/targetd.py:
```diff
@@ -148,6 +152,20 @@ def plugin_register(self, uri, password, timeout, flags=0):
+
+        if "cert_file" in self.uri["parameters"]:
+            # Check for file existence and throw error now if not present
+            self.cert_file = self.uri["parameters"]["cert_file"]
```
Can we raise `INVALID_ARGUMENT` error for when both `no_ssl_verify` and `cert_file` is used?
`targetd+ssl://localhost?cert_file=<file_path>&no_ssl_verify=yes`

tasleson replied:
Done

cathay4t commented on plugin/targetd/targetd.py:
```diff
@@ -148,6 +152,20 @@ def plugin_register(self, uri, password, timeout, flags=0):
+            self.cert_file = self.uri["parameters"]["cert_file"]
+
+            if not (os.path.exists(self.cert_file) and
+                    os.path.isfile(self.cert_file)):
```
`os.path.isfile` also check the existence: https://docs.python.org/2/library/os.path.html#os.path.isfile

cathay4t commented:
The `create_default_context` is new in python 2.7.9, we might need to enforce that in `configure.ac` or implement a workaround for python 6 or raise `INVALID_ARGUMENT` when `cert_file` is used in python 2.6 system.

tasleson commented:
@cathay4t Well only the newer versions of python probably do the certificate checking, so we should be able to maintain the old behavior, but I'll see what I can do to enable the same new features across all supported versions.

tasleson commented:
For newer versions of python, all our plugins that use urllib/urlib2 library probably need the same set of changes to support disable and adding cert file.

tasleson commented:
@cathay4t I've added the suggestions you made.  I also cleaned up the error handling for connection based errors.  Testing on F25 (py3) and EL7 & 6 (py2)

cathay4t commented:
Code looks good.
All manual tests passed:
 * `no_ssl_verify=yes`
 * `cert_file` conflicting with `no_ssl_verify`
 * `cert_file` with incorrect hostname.
 * Correct `cert_file`.

cathay4t commented:
Merged. Thank you Tony!
